server/src/main/java/org/opensearch/OpenSearchException.java

```diff
@@ -258,7 +259,10 @@ protected Map<String, List<String>> getHeaders() {
      */
     public RestStatus status() {
         Throwable cause = unwrapCause();
-        if (cause == this) {
+        if (cause.getCause() instanceof TaskCancelledException
+            && ((TaskCancelledException) cause.getCause()).status() == RestStatus.TOO_MANY_REQUESTS) {
+            return ((TaskCancelledException) cause.getCause()).status();
```

Comment:
One alternate here is to make it generic and throw the underlying cause's status in all cases till level 2 . We need to think about whether it is the right behavior and also make sure it doesn't cause any regression in existing code (ITs would make sure of that).  Something on the below lines. 

```
    public RestStatus status() {
        Throwable cause = unwrapCause();
        if (cause == this) {
            return RestStatus.INTERNAL_SERVER_ERROR;
        } else {
            if (cause.getCause() != cause) {
                return ExceptionsHelper.status(cause.getCause());
            }
            return ExceptionsHelper.status(cause);
        }
    }
```

Reply:
trying this

Reply:
The other better option is to override this in `TaskCancelledException` .

Reply:
@PritLadani Can you explore the second suggestion to do the override in TaskCancelledException? The code snippet above doesn't look quite right because it calls an "unwrapCause()" method but then proceeds to do another level of cause unwrapping. I'm really concerned about unintended side effects of that approach, and I also don't have a ton of confidence that ITs would catch every possible regression.